### PR TITLE
Max background jobs

### PR DESCRIPTION
**Pull request by DorianZheng**
## What have you changed? (mandatory)

raftdb support modify max_background_jobs

## What are the type of the changes? (mandatory)

Improvement

## How has this PR been tested? (mandatory)

test case

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

### CODE REVIEW

**Comment by breeswish** [comment]
Also remember to modify tidb-ansible :P

File: src/config.rs [hunk 1]
```diff
@@ -599,6 +600,7 @@ impl Default for RaftDbConfig {
             wal_ttl_seconds: 0,
             wal_size_limit: ReadableSize::kb(0),
             max_total_wal_size: ReadableSize::gb(4),
+            max_background_jobs: 6,
```
Comment (zhangjinpeng1987) [inline src/config.rs]: Is 6 is too large for raftdb? Since the total dataset is not large for this db.

Comment (siddontang) [reply src/config.rs]: seem they both use the same background pool?

Comment (zhangjinpeng1987) [reply src/config.rs]: Yes, so i'm wondering if this can make the kv-db's compaction starve for a while.

**Comment by DorianZheng** [comment]
> Please add it to config-template.toml.
`config-template.toml` already has this option

File: src/config.rs [hunk 2]
```diff
@@ -659,6 +662,7 @@ impl RaftDbConfig {
         opts.add_event_listener(EventListener::new("raft"));
         opts.set_bytes_per_sync(self.bytes_per_sync.0 as u64);
         opts.set_wal_bytes_per_sync(self.wal_bytes_per_sync.0 as u64);
+        // TODO maybe create a new env for raft engine
```
[the remark this answers is not in the document]
Comment (hicqu) [reply src/config.rs]: So we should set the default value of `max_background_jobs` of `RaftDbConfig` smaller?

[the remark this answers is not in the document]
Comment (zhangjinpeng1987) [reply src/config.rs]: I prefer the second one.

**Comment by hicqu** [comment]
LGTM.

**Pull request merged**